Completely changes the structure of the lib

zamith opened the pull request:
Due to my misunderstanding of the API, I was going with a solution which
realied greatly in class methods. Turns out that we're going to need the
user's token for each request, and it is probably going to be different
at each request.

This means that we need a Bitreserve::Client class that we instantiate
with that token and go on from there.

My approach was to make the client have a huge public API with all of
the methods available (provided their are being mixed in).

I also moved those mixins into a directory called api and moved the
actual classes into a directory called entities.

zamith committed:
Completely changes the structure of the lib

Due to my misunderstanding of the API, I was going with a solution which
realied greatly in class methods. Turns out that we're going to need the
user's token for each request, and it is probably going to be different
at each request.

This means that we need a Bitreserve::Client class that we instantiate
with that token and go on from there.

My approach was to make the client have a huge public API with all of
the methods available (provided their are being mixed in).

I also moved those mixins into a directory called api and moved the
actual classes into a directory called entities.

zamith commented:
@naps62 The build is breaking because I'm using keyword arguments, which ruby 1.9.3 does not support. I really love keyword arguments and would be willing to drop 1.9.3 support on that fact alone.

However, there are other reasons for dropping 1.9.3 support, such as the [Hampton's Ruby Survey](http://www.ruby-survey.com/stats#question_15) that says almost 80% of people have moved on to ruby 2, and the fact that 1.9.3 will be [EOL in less than a month](https://www.ruby-lang.org/en/news/2014/01/10/ruby-1-9-3-will-end-on-2015/).

What do you say?

naps62 commented:
Given those numbers, I don't have a problem with dropping 1.9.

Actually (and just for the record), Semaphore's numbers from 6 months ago show that more than 40% of ruby projects were running 1.9 (https://semaphoreapp.com/blog/2014/08/27/ruby-version-usage-in-commercial-projects-2014-edition.html)

So, assuming both statistics are accurate, that's a significant decrease in the last 6 months, which is also a pretty good number

zamith commented:
That survey is not comprehensive, but it does show trends. IMO the EOL coming is a bigger reason to change. I'll remove it from the test matrix, then.

zamith committed:
Drops support for Ruby 1.9.3

Most people have now stopped using Ruby 1.9.x, and it will be EOLed
shortly, so we decided not to support it.

zamith commented:
@naps62 What about the new structure?

naps62 commented:
:+1:
